packages/anthropic/src/anthropic-messages-language-model.ts

```diff
@@ -483,28 +475,25 @@ export class AnthropicMessagesLanguageModel implements LanguageModelV3 {
 
     const content: Array<LanguageModelV3Content> = [];
     const mcpToolCalls: Record<string, LanguageModelV3ToolCall> = {};
+    let isJsonToolCalled = false;
 
     // map response content to content array
     for (const part of response.content) {
       switch (part.type) {
         case 'text': {
-          // when a json response tool is used, the tool call is returned as text,
-          // so we ignore the text content:
-          if (!usesJsonResponseTool) {
-            content.push({ type: 'text', text: part.text });
-
-            // Process citations if present
-            if (part.citations) {
-              for (const citation of part.citations) {
-                const source = createCitationSource(
-                  citation,
-                  citationDocuments,
-                  this.generateId,
-                );
-
-                if (source) {
-                  content.push(source);
-                }
+          content.push({ type: 'text', text: part.text });
```

Comment:
this is a larger problem and a potential dealbreaker. I remember this is why tools are currently not supported with json output. Some anthropic models return text independent of tool calling, before the tool call. When that happens all output parsing is broken.

This needs to be reverted to the previous version, when structured output is active all text must be ignored.